contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1419,6 +1433,58 @@ get_db_owner_name(const char *dbname)
 	return get_db_owner_name_by_mode(dbname, get_migration_mode());
 }
 
+char *
+get_db_datareader_name_by_mode(const char *dbname, MigrationMode mode)
+{
+	char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
```

Comment:
Since this function always returns palloc'ed string, let's make sure to free the object wherever we have called this function.